### PR TITLE
Let the tutor authorize minors

### LINKED ISSUE #13: g9.6 Verify minor accounts through the tutors' interface
As part of #6 and after implementing #12 (related to #16 and #17), in order for a minor account to be fully operation it must be verified.

The purpose of this task is to allow to verify any of the accounts listed in my "minor accounts" zone.
Implementing the last part of this diagram:
create_minor_account

- [x] After creating a blocked user account, and the minor has confirmed the email. The tutor needs to verify the account using the method specified int the system configuration (see #9 ) `minors_authorization`. Create a controller for authorizing a user that is not the logged one (maybe it is possible to reuse the existing verification controller, investigate that).
- [ ] If the minor is verified successfully, unblock the account.
- [ ] If the minor is verified successfully, send a welcome message for the minor (ie: you can now start using the platform) and another one for the tutor (ie: your minor account is fully operative now) (specific text to put in place will come later).

### PR DESCRIPTION
**Pull request by microstudi**
fixes #13 

- [x] Custom controller for handling Direct verification methods (2-steps verification methods are not supported for minors)
- [x] Checks the age before authorizing if `minor_authorization_age_attributes` is configured.
- [x] Test authorization controller

Screenshot from 2022-10-28 14-01-09
Screenshot from 2022-10-28 14-00-32

### CODE REVIEW

File: app/controllers/concerns/decidim/kids/needs_tutor_authorization.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        def permission_class_chain
+          [::Decidim::Kids::Permissions] + super
+        end
+
+        private
+
+        def tutor_verified?
+          @tutor_verified ||= begin
+            authorization = granted_authorizations(current_user).where(name: current_organization.tutors_authorization)
+            authorization.any? && !authorization.first.expired?
```
Comment (fblupi) [inline app/controllers/concerns/decidim/kids/needs_tutor_authorization.rb]: We could use the `minor_authorization` method to perform a "find_by" instead of using "where + first"

```suggestion
            authorization = minor_authorization(current_user)
            authorization && !authorization.expired?
```

File: app/models/concerns/decidim/kids/user_override.rb [hunk 2]
```diff
@@ -25,6 +25,13 @@ def minor?
         def tutor?
           minor_accounts.exists?
         end
+
+        def minor_age
+          return unless minor?
+          return if minor_data_birthday.blank?
+
+          Time.zone.today.year - minor_data_birthday.year
```
Comment (fblupi) [inline app/models/concerns/decidim/kids/user_override.rb]: If the minor birthday is in December, we will get one more year than expected in the other months.

I've found [here](https://medium.com/@craigsheen/calculating-age-in-rails-9bb661f11303) an easy way to do it:

```suggestion
          ((Time.zone.now - minor_data_birthday.to_time) / 1.year.seconds).floor
```

File: app/views/decidim/kids/authorizations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<% add_decidim_page_title(t("menu", scope: "decidim.kids.user")) %>
+<% content_for(:subtitle) { t("menu", scope: "decidim.kids.user") } %>
+
+<div class="callout secondary">
+<%= minor_user.name %>
+<%= link_to "cancel", :back, class: "button secondary small float-right" %>
+</div>
+      <div class="columns large-8 large-centered text-center page-title">
```
Comment (fblupi) [inline app/views/decidim/kids/authorizations/new.html.erb]: We should fix the indentation in this file

File: app/commands/decidim/kids/authorize_user.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+      private
+
+      def valid_age?
+        return true unless handler.valid?
+        return true if Decidim::Kids.minor_authorization_age_attributes.blank?
+
+        Decidim::Kids.minor_authorization_age_attributes.detect do |attr|
+          begin
+            age = Time.zone.today.year - Date.parse(handler.try(attr).to_s).year
```
Comment (fblupi) [inline app/commands/decidim/kids/authorize_user.rb]: If the minor birthday is in December, we will get one more year than expected in the other months.

I've found [here](https://medium.com/@craigsheen/calculating-age-in-rails-9bb661f11303) an easy way to do it:

```suggestion
            age = ((Time.zone.now - Date.parse(handler.try(attr).to_s).to_time) / 1.year.seconds).floor
```

**Review by fblupi — CHANGES_REQUESTED** [review]

File: app/controllers/concerns/decidim/kids/needs_tutor_authorization.rb (hunk 1 above)
Comment (microstudi) [reply app/controllers/concerns/decidim/kids/needs_tutor_authorization.rb]: unfortunately this doesn't work because we need to check the tutor's authorization here, not the minor's. But I've rearrange the code.

File: app/models/concerns/decidim/kids/user_override.rb (hunk 2 above)
Comment (microstudi) [reply app/models/concerns/decidim/kids/user_override.rb]: Good point, added a spec too to check the edge case

File: app/commands/decidim/kids/authorize_user.rb (hunk 4 above)
Comment (microstudi) [reply app/commands/decidim/kids/authorize_user.rb]: added a test too

**Review by fblupi — APPROVED** [review]
LGTM 🚀
